docs/source/developer_tutorial.ipynb

```diff
@@ -0,0 +1,172 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "058b147f",
+   "metadata": {},
+   "source": [
+    "To begin to develop the `kinisi` code, you should clone the git repository and install `kinisi` with the `dev` option in editable mode. \n",
+    "\n",
+    "```console\n",
+    "$ git clone https://github.com/bjmorgan/kinisi.git\n",
+    "$ cd kinisi\n",
+    "$ pip install -e '.[dev]'\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c331029",
+   "metadata": {},
+   "source": [
+    "# DiffusionAnalyzer and JumpDiffusionAnalyzer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3799b256",
+   "metadata": {},
+   "source": [
+    "Within `kinisi` the main classes the end user will interact with are `diffusion_analyzer` and `jump_diffusion_analyzer`. \n",
+    "\n",
+    "These are typically called with 3 main methods depending on the type of trajectory these are:\n",
+    "\n",
+    "* `from_universe` - For MDAnalysis Universe\n",
+    "* `from_xdatcar` - For Pymatgen Structures\n",
+    "* `from_ase` - For ASE objects\n",
+    "\n",
+    "\n",
+    "\n",
+    "For instance, `DiffusionAnalyzer.from_xdatcar` can be called as in the Diffusion Coeffcienct from a VASP file workbook `vasp_d.ipynb` via"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "30b74559",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import scipp as sc\n",
+    "from pymatgen.io.vasp import Xdatcar\n",
+    "from kinisi.analyze import DiffusionAnalyzer\n",
+    "rng = np.random.RandomState(42)\n",
+    "\n",
+    "params = {'specie': 'Li',\n",
+    "          'time_step': 2.0 * sc.Unit('fs'),\n",
+    "          'step_skip': 50 * sc.Unit('dimensionless'),\n",
+    "          }\n",
+    "\n",
+    "\n",
+    "params['dimension']= 'xy'\n",
+    "\n",
+    "xd = Xdatcar('./example_XDATCAR.gz')\n",
+    "diff = DiffusionAnalyzer.from_xdatcar(xd, **params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4331a507",
+   "metadata": {},
+   "source": [
+    "`DiffusionAnalyzer` is a subclass of `Analyzer` and so upon its initialization `Analyzer` is initialized first, `DiffusionAnalyzer` inherits from the `Analyzer`.\n",
+    "\n",
+    "```python\n",
+    "\n",
+    "class DiffusionAnalyzer(Analyzer):\n",
+    "\n",
+    "```\n",
+    "\n",
+    "\n",
+    "The `from_` class method of the `DiffusionAnalyzer` calls the Analyzer class `_from_` method via a super(), i.e. within `from_universe`\n",
+    "\n",
+    "```python\n",
```

Comment:
Do you want to permalink to the relevant lines in the code? It might help the text flow a bit more.